### ISSUE TITLE
Player malfunctions after the device goes into sleep mode - Safari (macOS)

### ISSUE THREAD

**Issue by mxAlexG** [CoderLine/alphaTab#971]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

**Summary:**
After turning on the device from sleep mode, playback doesn't resume when api.play() or api.playPause() are called.

**Detailed:**
While using Safari and viewing the same example player on the website's main page or other ones, if you put your device in sleep mode and it lasts for more than few seconds in this state (~10 seconds) . Then, if you turn it back on, you will lose the ability to resume playback (whether you're using api.play() or api.playPause() makes no difference). The cursor will move slightly to the adjacent note to its right then reverts back to its initial position (highlighting the note indefinitely as well).

I tested this with Chrome and Firefox on a Windows device, an actual Android device and an emulator. All function properly with zero issues when going through this procedure. Unfortunately, I can't provide a video capture example.

### Expected Behavior

When the device goes into sleep mode and is turned back on, the player should restore or maintain its state and function as if there was no interruption to the process irrespective of the duration.

### Steps To Reproduce

1. Load any web page that has a displayed instance of alphaTab with a loaded score
2. When the page finishes loading, put the device in sleep mode or equivalent
3. Wait for an average of 10 seconds
4. Turn the device on
5. If the player has UI controls for playback, click on the resume button
6. If no UI Controls exist, invoke the function api.play() or api.playPause()
7. The behavior occurs described in "Current Behavior"

### Found in Version

1.2

### Platform

Web

### Environment

```markdown
- **macOS- BigSur**:
- **Safari 14**:
```

**Comment by mxAlexG** [comment]
Is there a workaround for this where you can restart the player or something along the lines of this?

**Comment by Danielku15** [comment]
Seems like an issue on iOS side with no good workaround because it is a bug in WebKit. Internally they move the AudioContext to a suspended mode without reactivating it correctly. Due to the corrupt state reactivating and resuming the context seems not working correctly. 

Some references I found around this issue.
https://bugs.webkit.org/show_bug.cgi?id=237878
https://bugs.webkit.org/show_bug.cgi?id=231105
https://github.com/playcanvas/engine/issues/3977
https://github.com/goldfire/howler.js/issues/1525

Will need to check if I can add some workaround for Safari. Like this one: https://developer.apple.com/forums/thread/658375

**Comment by Danielku15** [comment]
@mxAlexG As I do not have any Apple Devices for development and testing testing any bugfix is really tricky on my side. Cloud Testing Platforms like BrowserStack prevent devices to go in sleep mode because they need the devices active for allowing access to it. 

I incorporated some AudioContext suspend/resume handling changes of playcanvas/engine into alphaTab and uploaded a testbuild here: 

https://demo.alphatab.net/crm/ios/playground/control.html#

On all operating systems and browsers I tested I could still play the sound and also on the one iPad I could use for testing I could not see any problems. Therefore I would need to rely on your feedback for a fix.
